python/kserve/kserve/protocol/grpc/servicer.py

```diff
@@ -76,20 +77,28 @@ async def ModelMetadata(
     async def RepositoryModelLoad(
         self, request: pb.RepositoryModelLoadRequest, context
     ) -> pb.RepositoryModelLoadResponse:
-        response = await self._mode_repository_extension.load(name=request.model_name)
+        response = await self._mode_repository_extension.load(model_name=request.model_name)
         return pb.RepositoryModelLoadResponse(model_name=response["name"], isLoaded=response["load"])
 
     async def RepositoryModelUnload(
         self, request: pb.RepositoryModelUnloadRequest, context
     ) -> pb.RepositoryModelUnloadResponse:
-        response = await self._mode_repository_extension.unload(name=request.model_name)
+        response = await self._mode_repository_extension.unload(model_name=request.model_name)
         return pb.RepositoryModelUnloadResponse(model_name=response["name"], isUnloaded=response["unload"])
 
     async def ModelInfer(
         self, request: pb.ModelInferRequest, context: ServicerContext
     ) -> pb.ModelInferResponse:
         headers = to_headers(context)
-        response_body, _ = await self._data_plane.infer(body=request, headers=headers, model_name=request.model_name)
+
+        infer_inputs = [InferInput(name=input_tensor.name, shape=input_tensor.shape,
+                                   datatype=input_tensor.datatype, data=input_tensor.contents)
+                        for input_tensor in request.inputs]
+        infer_request = InferRequest(model_name=request.model_name, infer_inputs=infer_inputs,
```

Comment:
Construct the `InferRequest` from gRPC `ModelInferRequest` and pass to the model handlers, so user can program against the same type as REST, the framework handles the REST/gRPC decoding and encoding.